python/cugraph/structure/graph.py

```diff
@@ -565,945 +277,394 @@ def from_numpy_matrix(self, np_matrix):
-            return (ddf == n).any().any().compute()
-        if self.renumbered:
-            tmp = self.renumber_map.to_internal_vertex_id(cudf.Series([n]))
-            return tmp[0] is not cudf.NA and tmp[0] >= 0
-        else:
-            df = self.edgelist.edgelist_df[["src", "dst"]]
-            return (df == n).any().any()
+        self.Impl._nodes["all_nodes"] = cudf.Series(nodes)
 
-    def has_edge(self, u, v):
-        """
-        Returns True if the graph contains the edge (u,v).
-        """
-        if self.edgelist is None:
-            raise Exception("Graph has no Edgelist.")
-        if self.renumbered:
-            tmp = cudf.DataFrame({"src": [u, v]})
-            tmp = tmp.astype({"src": "int"})
-            tmp = self.add_internal_vertex_id(
-                tmp, "id", "src", preserve_order=True
-            )
+    # TODO: Add function
+    # def properties():
 
-            u = tmp["id"][0]
-            v = tmp["id"][1]
 
-        df = self.edgelist.edgelist_df
-        if self.distributed:
-            return ((df["src"] == u) & (df["dst"] == v)).any().compute()
-        return ((df["src"] == u) & (df["dst"] == v)).any()
+class DiGraph(Graph):
+    def __init__(self, m_graph=None):
+        warnings.warn(
+            "DiGraph is deprecated, use Graph(directed=True) instead",
+        )
+        super(DiGraph, self).__init__(m_graph, directed=True)
 
-    def edges(self):
-        """
-        Returns all the edges in the graph as a cudf.DataFrame containing
-        sources and destinations. It does not return the edge weights.
-        For viewing edges with weights use view_edge_list()
-        """
-        return self.view_edge_list()[["src", "dst"]]
 
-    def nodes(self):
-        """
-        Returns all the nodes in the graph as a cudf.Series
-        """
-        if self.distributed:
-            raise Exception("Not supported for distributed graph")
-        if self.edgelist is not None:
-            df = self.edgelist.edgelist_df
-            if self.renumbered:
-                # FIXME: If vertices are multicolumn
-                #        this needs to return a dataframe
-                # FIXME: This relies on current implementation
-                #        of NumberMap, should not really expose
-                #        this, perhaps add a method to NumberMap
-                return self.renumber_map.implementation.df["0"]
-            else:
-                return cudf.concat([df["src"], df["dst"]]).unique()
-        if self.adjlist is not None:
-            return cudf.Series(np.arange(0, self.number_of_nodes()))
-        if "all_nodes" in self._nodes.keys():
-            return self._nodes["all_nodes"]
-        else:
-            n = cudf.Series(dtype="int")
-            set_names = [i for i in self._nodes.keys() if i != "all_nodes"]
-            for k in set_names:
-                n = n.append(self._nodes[k])
-            return n
+class MultiGraph(Graph):
+    def __init__(self, directed=False):
```

Comment:
I am a bit confused by giving the user the option to instantiate a `MultiGraph` with the option `directed=True`. I was expecting something like 
```
    def __init__(self):
        super(MultiGraph, self).__init__(directed=True)
        self.graph_properties.multi_edge = True  #if multi_edge is an attribute of `Properties` class
```
or 
```
    def __init__(self):
        super(MultiGraph, self).__init__(directed=True, multi_edge=True) #if multi_edge is an attribute of `Properties` class
```
but I believe Brad said those attribute (multi_edge, etc) shouldn't be part of the Graph class

Reply:
The new API is deprecating explicit directed classes. So a MultiDiGraph needs to be instantiated as MultiGraph(directed =True) and it sets the multi_edge property